WordPress/Classes/ViewRelated/Site Creation/SiteSegments/SiteSegmentsWizardContent.swift

```diff
@@ -138,18 +144,43 @@ final class SiteSegmentsWizardContent: UIViewController {
     }
 
     private func fetchSegments() {
-        service.siteSegments(for: Locale.current) { [weak self] results in
+        guard isNetworkActive == true else {
+            setupEmptyTableProvider()
+            return
+        }
+
+        service.siteSegments(completion: { [weak self] results in
             switch results {
-            case .error(let error):
+            case .failure(let error):
                 self?.handleError(error)
             case .success(let data):
                 self?.handleData(data)
             }
-        }
+        })
+    }
+
+    private func handleError(_ error: SiteSegmentsError) {
+        setupEmptyTableProvider()
     }
 
-    private func handleError(_ error: Error) {
-        debugPrint("=== handling error===")
+    private func setupEmptyTableProvider() {
+        let message: InlineErrorMessage
+        if isNetworkActive {
+            message = InlineErrorMessages.serverError
+        } else {
+            message = InlineErrorMessages.noConnection
+        }
+
+        let handler: CellSelectionHandler = { [weak self] _ in
+            self?.fetchSegments()
+        }
+
+        dataCoordinator = InlineErrorTableViewProvider(tableView: table, message: message, selectionHandler: handler)
```

Comment:
I'm seeing two issues with this:

1. As built, the branch fails an assertion when the network is unreachable due to the fact that `InlineErrorRetryTableViewCell` is not properly registered.
1. We should not be using the "inline error" in this case; it's intended for search results ala Verticals & Domains.
1. I believe you'll want to use the `networkUnreachable` case of `ErrorStateViewType`; it might be helpful to refer to`SiteAssemblyWizardContent`.